apps/entity-service/src/entities/processors/media.processor.ts

```diff
@@ -225,21 +217,31 @@ export class MediaProcessor extends AssociationProcessor<Media, MediaDto> {
         entityType: this.entityType,
         entityUuid: this.entityUuid,
         url: this.entitiesService.fullUrl(media),
-        thumbUrl: this.entitiesService.thumbnailUrl(media)
+        thumbUrl: this.entitiesService.thumbnailUrl(media),
+        modelType: this.entityType
       };
 
       document.addData(association.uuid, new this.DTO(association, additionalProps));
     }
 
-    const total = await this.getTotal(await this.getBaseEntity(), query);
+    const total = await this.getTotal();
 
     const resource = getDtoType(this.DTO);
     document.addIndexData({
       resource,
-      requestPath: `/entities/v3/${this.entityType}/${this.entityUuid}/${resource}${getStableRequestQuery(query)}`,
+      requestPath: `/entities/v3/${this.entityType}/${this.entityUuid}/${resource}${getStableRequestQuery(this.query)}`,
       total,
-      pageNumber: query.page?.number,
+      pageNumber: this.query?.page?.number,
       ids: indexIds
     });
   }
+
+  _queryBuilder: Promise<PaginatedQueryBuilder<Media>> | null = null;
+
+  get queryBuilder() {
+    if (this._queryBuilder == null) {
+      this._queryBuilder = this.getQueryBuilder();
+    }
+    return this._queryBuilder;
+  }
```

Comment:
I don't love a getter that returns a promise - getters typically should return a resolved value. This is also not following previous patterns in this codebase. 

Instead, `getQueryBuilder()` should do the memoization itself and simply return the builder if it has already been built. See how this pattern is established in `getBaseEntity`

Reply:
changed